runtime/src/message_processor.rs

```diff
@@ -66,38 +66,39 @@ fn verify_instruction(
 ) -> Result<(), InstructionError> {
     // Verify the transaction
 
-    // Make sure that program_id is still the same or this was just assigned by the system program,
-    //  but even the system program can't touch a credit-only account.
-    if pre.owner != post.owner && (!is_debitable || !system_program::check_id(&program_id)) {
+    // Make sure that program_id is still the same or this was just re-assigned by
+    //  the owner program
+    if pre.owner != post.owner && (!is_debitable || *program_id != pre.owner) {
```

Comment:
Whoa this is new.   We want to enable any program to `chown`?  That seems dangerous, I could construct an account and then inject it into your program

Reply:
this PR is an open question

Reply:
the rationale is that programs only need to prepare for zero-initialized, but possibly wrong-sized buffers?

Reply:
nevermind, I get it.